lib/Doctrine/DBAL/Platforms/DB2Platform.php

```diff
@@ -771,16 +771,26 @@ public function getTemporaryTableName($tableName)
      */
     protected function doModifyLimitQuery($query, $limit, $offset = null)
     {
-        if ($limit === null && $offset === null) {
-            return $query;
+        $where = array();
+
+        if ($offset > 0) {
+            $where[] = sprintf('db22.DC_ROWNUM >= %d', $offset + 1);
         }
 
-        $limit = (int) $limit;
-        $offset = (int) (($offset)?:0);
+        if ($limit !== null) {
+            $where[] = sprintf('db22.DC_ROWNUM <= %d', $offset + $limit);
+        }
+
+        if (count($where) === 0) {
```

Comment:
@morozov this is just about minor optimizations. `empty()` is a language construct and much more efficient than using the function `count()`. The point about typos is a good point, but in DBAL we try to code as much performance efficient as possible even if it seems neglible. Thanks for changing.